packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -1432,8 +1411,8 @@ function createObservationSocialHistorySection(observations: Observation[]) {
           LOINC_CODE,
         ]),
         value,
-        firstDate: observationDate,
-        lastDate: observationDate,
+        firstDate: observation.effectivePeriod?.start ? firstDate : observationDate,
+        lastDate: observation.effectivePeriod?.end ? lastDate : observationDate,
```

Comment:
Its confusing (and error prone as we maintain this code) to check a variable and use another one.

Maybe do this instead?

```ts
  const firstDate = observation.effectivePeriod?.start;
  const lastDate = observation.effectivePeriod?.end;
  ...
  firstDate: firstDate ? formatDateForDisplay(firstDate) : observationDate,
  lastDate: lastDate ? formatDateForDisplay(lastDate) : observationDate,
```